nuxeo-features/preview/nuxeo-preview-jsf/src/main/java/org/nuxeo/ecm/platform/preview/restlet/PreviewRestlet.java

```diff
@@ -0,0 +1,302 @@
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jboss.seam.ScopeType;
+import org.jboss.seam.annotations.In;
+import org.jboss.seam.annotations.Name;
+import org.jboss.seam.annotations.Scope;
+import org.jboss.seam.international.LocaleSelector;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentNotFoundException;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.blobholder.BlobHolder;
+import org.nuxeo.ecm.core.api.blobholder.DocumentBlobHolder;
+import org.nuxeo.ecm.core.blob.BlobManager;
+import org.nuxeo.ecm.core.blob.BlobManager.UsageHint;
+import org.nuxeo.ecm.core.io.download.DownloadService;
+import org.nuxeo.ecm.platform.preview.adapter.base.ConverterBasedHtmlPreviewAdapter;
+import org.nuxeo.ecm.platform.preview.api.HtmlPreviewAdapter;
+import org.nuxeo.ecm.platform.preview.api.NothingToPreviewException;
+import org.nuxeo.ecm.platform.preview.api.PreviewException;
+import org.nuxeo.ecm.platform.ui.web.api.NavigationContext;
+import org.nuxeo.ecm.platform.ui.web.restAPI.BaseNuxeoRestlet;
+import org.nuxeo.ecm.platform.util.RepositoryLocation;
+import org.nuxeo.ecm.platform.web.common.locale.LocaleProvider;
+import org.nuxeo.ecm.webapp.helpers.ResourcesAccessor;
+import org.nuxeo.runtime.api.Framework;
+import org.restlet.data.MediaType;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+import org.restlet.data.Status;
+
+/**
+ * Provides a REST API to retrieve the preview of a document.
+ *
+ * @author tiry
+ * @since 10.3
+ * @deprecated since 10.3
+ */
+@Name("previewRestlet")
+@Scope(ScopeType.EVENT)
+public class PreviewRestlet extends BaseNuxeoRestlet {
+
+    private static final Log log = LogFactory.getLog(PreviewRestlet.class);
+
+    public static final String PREVIEWURL_DEFAULTXPATH = "default";
+
+    @In(create = true)
+    protected NavigationContext navigationContext;
+
+    protected CoreSession documentManager;
+
+    protected DocumentModel targetDocument;
+
+    @In(create = true)
+    protected transient LocaleSelector localeSelector;
+
+    @In(create = true)
+    protected transient ResourcesAccessor resourcesAccessor;
+
+    // cache duration in seconds
+    // protected static int MAX_CACHE_LIFE = 60 * 10;
+
+    // protected static final Map<String, PreviewCacheEntry> cachedAdapters =
+    // new ConcurrentHashMap<String, PreviewCacheEntry>();
```

Comment:
Remove commented code?

Reply:
Took back the `PreviewRestlet` class as it from `8.10` also.